### ISSUE TITLE
Lazy sequence is reversed in coercion

### ISSUE THREAD

**Issue by salokristian** [metosin/spec-tools#176]
# Library version
0.9.1

# Problem
When a lazy seq is given to `st/coerce`, it is reversed. 

Given a `coll-of` spec
```
(s/def ::coll-spec (s/coll-of int?))
```

And a lazy seq as input
```
(st/coerce ::coll-spec (lazy-seq '(1 2 3)) st/string-transformer)
=> (3 2 1)
```

The resulting list is reversed.

**Comment by salokristian** [comment]
I can also look into this myself this weekend.

**Comment by jrasanen** [comment]
Thanks, I think we fixed that for the [0.9.0 version](https://github.com/metosin/spec-tools/commit/cb3e22baea68772f9ca99a963984975c8112541b), but maybe it's a regression or we missed some case.
